pkgs/by-name/ni/nixos-render-docs/src/nixos_render_docs/manual.py

```diff
@@ -318,6 +321,7 @@ def _file_header(self, toc: TocEntry) -> str:
                      for style in self._html_params.stylesheets)),
             "".join((f'<script src="{html.escape(script, True)}" type="text/javascript"></script>'
                      for script in self._html_params.scripts)),
+            f'<script type="text/javascript">{self._redirects.get_redirect_script(toc.target.path)}</script>',
```

Comment:
This embeds `2.3M` of text into the index.html. Potentially more in the future, when people start to extend the map file.

Better:

`<script type="module" src="path/to/script.js"></script>` (generate a file, instead of inlining 2.3 MB, which is pretty bad for loading time)

This allows the browser to load the script seperate from the content. Javascript-modules are supported across all major browser. This also gives us the possibility in the future to only load the file if we detect a user coming from an old anchor.

Reply:
It does not embed `2.3M` of redirects into `index.html`. We only send in redirects that actually exist, not what you see in `redirects.json`. See [`get_client_redirects()`](https://github.com/GetPsyched/nixpkgs/blob/dca60b3027d630732e2670b92b7cc13622c0939b/pkgs/by-name/ni/nixos-render-docs/src/nixos_render_docs/redirects.py#L143-L154)

That said, I agree with you on the inline vs. module JS suggestion, I will make that change.